app/src/controllers/object.js

```diff
@@ -249,6 +300,109 @@ const controller = {
     }
   },
 
+  /**
+   * @function deleteMetadata
+   * Creates a new version of the object via copy with the given metadata removed
+   * @param {object} req Express request object
+   * @param {object} res Express response object
+   * @param {function} next The next callback function
+   * @returns {function} Express middleware function
+  */
+  async deleteMetadata(req, res, next) {
+    try {
+      const objId = addDashesToUuid(req.params.objId);
+      const objPath = getPath(objId)
+
+      const latest = await storageService.headObject({ filePath: objPath });
+      if (latest.ContentLength > 5368709120) {
+        throw new Error('Cannot copy an object larger than 5GB');
+      }
+      else {
+        const keysToRemove = mixedQueryToArray(req.query.key);
+        let metadata = latest.Metadata;
+
+        if (keysToRemove.hasOwnProperty('versionId')) delete keysToRemove['versionId'];
+
+        if (keysToRemove !== undefined && keysToRemove.length > 0) {
+          keysToRemove.forEach(x => {
+            if (metadata.hasOwnProperty(x)) delete metadata[x];
```

Comment:
You may be able to simplify the logic here by doing it decoratively instead of imperatively using ES2019 style code:
```js
// Generate object subset by subtracting/omitting defined keys via filter/inclusion
const newMetadataSet = Object.fromEntries(
  Object.entries(metadata)
    .filter(([key]) => !keysToRemove.includes(key))
)
```
As a side note, this syntax is all supported in Node.js 12.x and above, so we're good.
Ref: https://stackoverflow.com/a/56592365